### ISSUE TITLE
OCaml 5.2 support

### ISSUE THREAD

**Issue by kit-ty-kate** [stedolan/malfunction#40]
malfunction currently fails with the following error on OCaml 5.2:
```
#=== ERROR while compiling malfunction.0.5 ====================================#
# context              2.2.0~beta2~dev | linux/x86_64 | ocaml-variants.5.2.0+trunk | file:///home/opam/opam-repository
# path                 ~/.opam/5.2/.opam-switch/build/malfunction.0.5
# command              ~/.opam/5.2/bin/dune build -p malfunction -j 1
# exit-code            1
# env-file             ~/.opam/log/malfunction-20-93572e.env
# output-file          ~/.opam/log/malfunction-20-93572e.out
### output ###
# (cd _build/default && /home/opam/.opam/5.2/bin/ocamlc.opt -w -40 -g -bin-annot -I src/.malfunction.objs/byte -I /home/opam/.opam/5.2/lib/findlib -I /home/opam/.opam/5.2/lib/ocaml/compiler-libs -I /home/opam/.opam/5.2/lib/ocaml/dynlink -I /home/opam/.opam/5.2/lib/ocaml/str -I /home/opam/.opam/5.2/lib/ocaml/unix -I /home/opam/.opam/5.2/lib/zarith -no-alias-deps -o src/.malfunction.objs/byte/malfunction_compat.cmo -c -impl src/malfunction_compat.ml)
# File "malfunction_compat.cppo.ml", lines 15-25, characters 13-3:
# Error: Some record fields are undefined: "may_fuse_arity"
```

**Comment by stedolan** [comment]
Thanks for the report, having a look at this.

**Comment by stedolan** [comment]
Fixed in v0.6
